pkg/resources/kafka/pod.go

```diff
@@ -404,15 +405,24 @@ func generateVolumeMountForSSL() []corev1.VolumeMount {
 	}
 }
 
-func generateEnvConfig(brokerConfig *v1beta1.BrokerConfig, defaultEnvVars, clusterEnvVars []corev1.EnvVar) []corev1.EnvVar {
+func generateEnvConfig(brokerConfig *v1beta1.BrokerConfig, defaultEnvVars []corev1.EnvVar) []corev1.EnvVar {
 	envs := map[string]corev1.EnvVar{}
 
 	for _, v := range defaultEnvVars {
 		envs[v.Name] = v
 	}
 
-	for _, v := range clusterEnvVars {
-		envs[v.Name] = v
+	// merge the env variables
+	for _, envVar := range brokerConfig.Envs {
+		if envVar.Value != "" && envVar.Value[0] == '+' {
```

Comment:
do we strip this values somewhere? whitespaces could be a real paint in the a...